kilometer-backend-domain/src/main/java/com/kilometer/domain/user/User.java

```diff
@@ -87,7 +88,11 @@ public User update(UserUpdateRequest userUpdateRequest) {
         this.email = userUpdateRequest.getEmail();
         this.gender = Gender.valueOf(userUpdateRequest.getGender());
         this.phoneNumber = userUpdateRequest.getPhoneNumber();
-        this.birthdate = userUpdateRequest.getBirthDay().atStartOfDay();
+        if (Objects.nonNull(userUpdateRequest.getBirthDay())) {
+            this.birthdate = userUpdateRequest.getBirthDay().atStartOfDay();
+        } else {
+            this.birthdate = null;
+        }
```

Comment:
`birthDay`의 null 검증을 메서드로 분리하는 것은 어떨까요?!

객체지향 생활 체조 원칙(else 표현을 사용하지 않는다)
else문을 제거하는것은 어떨까요?!
가볍게 여쭤보아요!!

Reply:
> `birthDay`의 null 검증을 메서드로 분리하는 것은 어떨까요?!
> 
> 객체지향 생활 체조 원칙(else 표현을 사용하지 않는다) else문을 제거하는것은 어떨까요?! 가볍게 여쭤보아요!!
이런 원칙도 있군여?

Reply:
이 부분은 제가 else 표현을 사용하지 않는다 라는 걸 봤는데, 연속적인 환경에서 조기 return시켜서 가독성들을 높이는 것을 의미하는데, 여기서는 관련이 없는 것 같습니다.

여기서 else는 실제로 null로 들어오면 기존의 데이터를 삭제해야하기에 만약 else를 없앤다면 이런 코드가 될거 같아요.
```suggestion
        if (Objects.nonNull(userUpdateRequest.getBirthDay())) {
            this.birthdate = userUpdateRequest.getBirthDay().atStartOfDay();
        } 
        if(Objects.isNull(userUpdateRequest.getBirthDay()))
            this.birthdate = null;
        }
```
이런 코드를 원하시는 건 아닐거라 생각하는데 어떤 걸 원하시는걸까요?

Reply:
``` JAVA
public void update(final serUpdateRequest request) {
        this.birthdate = birthDay(request.getBirthDay());
}

private String birthday(final serUpdateRequest request) {
        if (Objects.nonNull(userUpdateRequest.getBirthDay())) {
                return userUpdateRequest.getBirthDay().atStartOfDay();
        }
        return null;
}
```

현재 구조에서는 위와 같이 변경해 볼수 있을것 같아요!
`update()` 메서드의 목적은 update를 하기 위함이기에 검증로직을 분리해줄 수 있다고 생각하는데요
위와같이 `birthday()` 메서드를 통해 분리해줄수도 있을것 같아요 더 정확히는 
``` java
public void update(final serUpdateRequest request) {
        this.birthdate = birthDay(request.getBirthDay());
}

private String birthday(final serUpdateRequest request) {
        if (Objects.nonNull(userUpdateRequest.getBirthDay())) {
                return userUpdateRequest.getBirthDay().atStartOfDay();
        }
        throw new IllegalArgumentException();
}
```
과 같이 null이면 예외가 발생하도록 해줄수도 있을것 같아요

객체지향 생활체조에서 else를 지양하는 원칙을 세운 이유는 `if - else if - else` 절로 이어지는 부분들을 `메서드로 분리`하여
객체간 협력 관계, 메서드 의도에 따른 분리를 잘하게끔 만들기 위함이라고 생각하는데요

또한 else를 사용할 경우 예외가 발생하는 지점에서도 정상경우로 통과시키기에 버그발생가능성이 높아진다고 생각해요.
그리고 성능면에서도 `메서드 분리`를 하면 이점이 있다고 생각하는데요! 
메서드 분리를 잘 해 놓을수 록 JVM의 JIT 컴파일러가 성능 최적화를 하는데 도움이 되기 때문이예요!

다음과 같은 이유로 저는 else 구문을 지양하는 편입니다!